src/arrayMethodSort.js

```diff
@@ -4,8 +4,22 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  const defaultCompare = function defaultCompare(a, b) {
+    return a.toString() > b.toString();
+  };
+
+  [].__proto__.sort2 = function (compareFunction = defaultCompare) {
+    const compare = compareFunction;
+
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let j = 0; j < this.length - 1 - i; j++) {
+        if (compare(this[j], this[j + 1]) > 0) {
+          [this[j], this[j + 1]] = [this[j + 1], this[j]];
```

Comment:
The swap operation is correct, but keep in mind that using array destructuring for swapping elements in a sort algorithm may not be the most efficient approach due to the additional array creation overhead. In performance-critical code, a temporary variable might be a better choice.